src/Evolution/Systems/Cce/Events/ObserveFields.hpp

```diff
@@ -0,0 +1,421 @@
+#include "Time/Tags/Time.hpp"
+#include "Utilities/MakeString.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TypeTraits/IsA.hpp"
+
+/// \cond
+template <size_t Dim>
+class Mesh;
+namespace Frame {
+struct Inertial;
+}  // namespace Frame
+/// \endcond
+
+namespace Cce {
+namespace Events {
+namespace detail {
+template <typename Tag>
+std::string name() {
+  if constexpr (std::is_same_v<Tag, Tags::ComplexInertialRetardedTime>) {
+    return db::tag_name<Tags::InertialRetardedTime>();
+  } else {
+    return db::tag_name<Tag>();
+  }
+}
+}  // namespace detail
+
+/*!
+ * \brief Event to observe fields/variables in a characteristic evolution.
+ *
+ * \details Similar to `dg::Events::ObserveFields`, this event will write volume
+ * data from the characteristic domain to disk when triggered. However, there a
+ * several differences which are important to highlight.
+ *
+ * First is the fields themselves. The DG event takes the fields to observe as
+ * template parameters because the event must work with many evolution systems.
+ * However, since this event is specific to the characteristic evolution system,
+ * we can hardcode the list of fields that are available to observe. The fields
+ * available to observe are the following tags along with their first and second
+ * `Cce::Tags::Dy` derivatives (see `Cce::Tags::Dy` for a definition of y`):
+ *
+ * - `Cce::Tags::BondiBeta`
+ * - `Cce::Tags::BondiU`
+ * - `Cce::Tags::BondiQ`
+ * - `Cce::Tags::BondiW`
+ * - `Cce::Tags::BondiH` (no second derivative)
+ * - `Cce::Tags::BondiJ`
+ * - `Cce::Tags::Du<Cce::Tags::BondiJ>`
+ *
+ * Some more fields to observe are:
+ *
+ * - `Cce::Tags::ComplexInertialRetardedTime`
+ * - `Cce::Tags::OneMinusY`
+ * - `Cce::Tags::BondiR`
+ * - `Cce::Tags::EthRDividedByR`
+ * - `Cce::Tags::DuRDividedByR`
+ *
+ * The main reason that this event is separate from the DG one is because this
+ * event writes modal data over the sphere for every radial grid point, while
+ * the DG event writes nodal data. Every tag above is a
+ * `Scalar<SpinWeighted<ComplexDataVector, Spin>>` for some `Spin`. While this
+ * data itself is in nodal form, it is more convenient to transform to modal
+ * data and decompose in spherical harmonics before writing. This means
+ * our typical way of writing/storing volume data won't work.
+ *
+ * This event writes its data in the following structure in the H5 file:
+ * `/Cce/VolumeData/TagName/CompactifiedRadius_X.dat`. Every field that is
+ * observed will get its own subgroup called `TagName`. In this subgroup, there
+ * will be N files corresponding to N radial grid points named
+ * `CompactifiedRadius_X.dat` where `X` here will range from 0 to `N-1`. We call
+ * these compactified radii because for a more "physical" radius, it goes to
+ * infinity at future-null infinity and we can't write that in a file. Instead,
+ * these N files will correspond to the compactified coordinate $y = 1 - 2R/r$
```

Comment:
Your equations aren't rendering correctly.  I don't see an obvious cause.

Reply:
This rendered locally for me once I fixed the missing back-tick above.

Reply:
Yep, that did it.